### PR TITLE
Support Binance Smart Chain

### PR DESCRIPTION
**Pull request by FitzLu**
- Query transaction history
- Parse BEP20 token from contract
- Parse crosschain transfer

### CODE REVIEW

File: server/public.go [hunk 1]
```diff
@@ -998,6 +1001,75 @@ func (s *PublicServer) apiAddress(r *http.Request, apiVersion int) (interface{},
...
+	info, err := s.chain.EthereumTypeGetErc20ContractInfo(addrDesc)
+	if err != nil {
+		return nil, api.NewAPIError("fetch contract info failed", true)
+	}
+
+	if info == nil {
+		return nil, api.NewAPIError("fail to fetch token info", true)
+	}
+
+	iconUrl := fmt.Sprintf("https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/%s/assets/%s/logo.png", s.chain.GetCoinName(), address)
```
Comment (vikmeup) [inline server/public.go:1060]: Icon url could be constructed from outside of blockbook response.

File: server/public.go [hunk 2]
```diff
@@ -998,6 +1001,75 @@ func (s *PublicServer) apiAddress(r *http.Request, apiVersion int) (interface{},
...
+	erc20Infos := make([]*bchain.Erc20ContractInfo, 0)
+	for _, t := range tokens {
+		if len(t.Contract) > 0 {
+			var iconUrlHandle string
+			if s.chain.GetCoinName() == "BSC" {
+				iconUrlHandle = "bsc"
+			} else if s.chain.GetCoinName() == "BSCTestnet" {
+				iconUrlHandle = "bsct"
+			}
+			iconUrl := fmt.Sprintf("https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/%s/assets/%s/logo.png",
```
Comment (vikmeup) [inline server/public.go:1019]: same here

File: bchain/types.go [hunk 3]
```diff
@@ -210,6 +217,14 @@ type Erc20Contract struct {
 	Decimals int    `json:"decimals"`
 }
 
+type Erc20ContractInfo struct {
```
Comment (vikmeup) [inline bchain/types.go:220]: You no longer need `Erc20ContractInfo`, use `Erc20Contract`

**Comment by martinboehm** [comment]
Hi. You have done an impressive amount of work.

Unfortunately, the code you have created significantly diverges from our code, therefore we cannot merge your pull request.
I would recommend you to keep a separate fork, in which you can implement the changes required for the coins you want to support.

**Pull request closed without merging**